schemas/schema.gql

```diff
@@ -1,14 +1,65 @@
-type communities {
-  name: String!
+type Member {
+  username: String!
   address: String!
+  ens: String!
+  community: [Community!] @relation
+}
+
+type Integration {
+  cid: String!
+  community: Community!
+}
+
+type Skillset {
+  name: String!
+  description: String
+  community: Community!
+  skills: [Skill!] @relation
+}
+
+type Skill {
+  name: String!
+  description: String
+  skillsets: [Skillset!] @relation
+}
+
+type RollupConfig {
+  systemName: String!
+  skill: Skill!
+  functionName: String!
+  dataSourcesKeys: [String!]
+  community: Community!
+}
+
+type RollupCache {
   cid: String!
-  members: {
-    username: String!
-    address: String!
-    ens: String!
-  }
+  rollupData: String!
+  community: Community!
+}
+
+type Community {
+  name: String!
+  address: String!
+  ens: String
+  members: [Member!] @relation
+  integrations: [Integration!] @relation
+  skillset: [Skillset!] @relation
+  rollupConfigs: [RollupConfig!] @relation
 }
 
 type Query {
-  allCommunities: [communities!]
+  allCommunities: [Community!]
+  community(address: String!): Community
+  createCommunity(name: String!, address: String!, ens: String): Community
+  allMembers: [Member!]
+  createMember(username: String!, address: String!, ens: String): Member
+  allIntegrations: [Integration!]
+  createIntegration(cid: String!): Integration
+  allSkills: [Skill!]
+  skill(name: String!): Skill
+  createSkill(name: String!): Skill
+  rollupConfig(functionName: String!): [RollupConfig!]
+  createRollupConfig(functionName: String!): RollupConfig
```

Comment:
This may need the skill, community and dataSourcesKeys as well as input?

you could also define input DTO as follows:

```graphql
input RollupConfigCreateInput {
  systemName: String!
  skillId: String!
  functionName: String!
  dataSourcesKeys: [String!]
  communityId: String!
}
```

Then, go with:

```graphql
createRollupConfig(data: RollupConfigCreateInput!): RollupConfig
```

Wondering how associations will work here, it looks like we could use something like [connect](https://docs.fauna.com/fauna/current/api/graphql/relationships#connect) to perform the skill and community associations.

Reply:
Nevermind! LOL we can adjust how fauna expects data, but, at least we need to take into consideration those other fields.

Reply:
Awesome! ok adding them, thanks!